lib/src/main/java/io/cloudquery/helper/ArrowHelper.java

```diff
@@ -15,29 +18,136 @@
 import java.util.Map;
 import org.apache.arrow.memory.BufferAllocator;
 import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.BigIntVector;
+import org.apache.arrow.vector.BitVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.FixedSizeBinaryVector;
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.Float8Vector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.LargeVarBinaryVector;
+import org.apache.arrow.vector.LargeVarCharVector;
+import org.apache.arrow.vector.SmallIntVector;
+import org.apache.arrow.vector.TimeStampVector;
+import org.apache.arrow.vector.TinyIntVector;
+import org.apache.arrow.vector.UInt1Vector;
+import org.apache.arrow.vector.UInt2Vector;
+import org.apache.arrow.vector.UInt4Vector;
+import org.apache.arrow.vector.UInt8Vector;
+import org.apache.arrow.vector.VarBinaryVector;
+import org.apache.arrow.vector.VarCharVector;
 import org.apache.arrow.vector.VectorSchemaRoot;
 import org.apache.arrow.vector.ipc.ArrowReader;
 import org.apache.arrow.vector.ipc.ArrowStreamReader;
 import org.apache.arrow.vector.ipc.ArrowStreamWriter;
 import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
 import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.arrow.vector.util.Text;
 
 public class ArrowHelper {
+  public static final String CQ_EXTENSION_INCREMENTAL = "cq:extension:incremental";
+  public static final String CQ_EXTENSION_CONSTRAINT_NAME = "cq:extension:constraint_name";
+  public static final String CQ_EXTENSION_PRIMARY_KEY = "cq:extension:primary_key";
+  public static final String CQ_EXTENSION_UNIQUE = "cq:extension:unique";
   public static final String CQ_TABLE_NAME = "cq:table_name";
   public static final String CQ_TABLE_TITLE = "cq:table_title";
   public static final String CQ_TABLE_DESCRIPTION = "cq:table_description";
   public static final String CQ_TABLE_DEPENDS_ON = "cq:table_depends_on";
 
+  private static void setVectorData(FieldVector vector, Object data) {
+    vector.allocateNew();
+    if (vector instanceof BigIntVector) {
+      ((BigIntVector) vector).set(0, (long) data);
+      return;
+    }
```

Comment:
You can follow this pattern which we do elsewhere to remove the cast on the line below:

```suggestion
    if (vector instanceof BigIntVector bigIntVector) {
      bigIntVector.set(0, (long) data);
      return;
    }
```